modules/appeals_api/app/services/appeals_api/evidence_submission_request_validator.rb

```diff
@@ -76,7 +78,7 @@ def invalid_review_option_error
     end
 
     def invalid_veteran_ssn_error
-      return unless appeal.auth_headers['X-VA-SSN']
+      return unless appeal.veteran&.ssn
```

Comment:
This relies on the updates to `Appellant` (`veteran` is an instance of `Appellant`)